### PR TITLE
feat(main): Add JSON schema for spec

### PR DESCRIPTION
**Pull request by yevgenypats**
Blocked by: https://github.com/cloudquery/plugin-sdk/pull/1214

### CODE REVIEW

**Review by disq — APPROVED** [review]

File: plugins/source/gcp/client/spec/spec.go [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
-	OrganizationIDs             []string           `json:"organization_ids"`
-	OrganizationFilter          string             `json:"organization_filter"`
-	ServiceAccountImpersonation *CredentialsConfig `json:"service_account_impersonation"`
 	Concurrency                 int                `json:"concurrency"`
 	Scheduler                   scheduler.Strategy `json:"scheduler,omitempty"`
+	ServiceAccountImpersonation *CredentialsConfig `json:"service_account_impersonation"`
+}
+
+func (spec *Spec) SetDefaults() {
+	if spec.BackoffRetries <= 0 {
```
Comment (bbernays) [inline plugins/source/gcp/client/spec/spec.go:29]: If a user wanted `0` retries how would they do that?

Comment (candiduslynx) [reply plugins/source/gcp/client/spec/spec.go]: Addressed in d426712e9bbb647b13eadb92186f57f17f0e6039

**Comment by hermanschaaf** [comment]
What do we expect to happen if we merge this PR? Will specs then be validated using JSONSchema, returning an error if validation fails?

**Comment by candiduslynx** [comment]
> What do we expect to happen if we merge this PR? Will specs then be validated using JSONSchema, returning an error if validation fails?

Yes, exactly

**Review by disq — APPROVED** [review]

**Pull request merged**